File: python/26_removeDuplicatesFromSortedArray.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        if len(nums) == 0:
            return len(nums)
        idx = 1
        last = nums[0]
        while idx < len(nums):
            if last == nums[idx]:
                del nums[idx]
            else:
                last = nums[idx]
                idx += 1
            
        return len(nums)

File: java/26_RemoveDuplicatesFromSortedArray.java
class Solution {
    public int removeDuplicates(int[] nums) {
        int left = 1, right = 1;
        
        while (right < nums.length) {
            if (nums[left] <= nums[left-1]) {
                while (right < nums.length && nums[right] == nums[right-1]) ++right;
                
                if (right < nums.length && nums[right] > nums[right-1])
                    nums[left++] = nums[right++];
            } else {
                ++left;
                ++right;
            }
        }
        return left;
    }
}
